WordPress/openverse
# Add structured logging for the API

## Fixes



Fixes #3432 by @sarayourfriend

## Description



This PR
- adds structured logging support for Django using 
- refactors the logging setup to remove cruft and dead code
- removes `django-log-request-id` as that functionality is included in `django-structlog`
- updates all loggers to use `structlog`

Please don't be daunted by the file count, most of these are `logging` &rarr; `structlog` migrations. These changes was required to make the tests pass, else I would've made them a separate PR. The main files to review are the 5 or so changed files inside `api/conf/`.

The PR is marked as 🟧 high-priority because it blocks all the work on monitoring metrics associated with content moderation. 

## Testing Instructions


0. Check out the branch and run the dev server: `just api/up`.
1. Tail the server logs: `just logs web`.
2. Access some endpoints, perform some tasks and observe these points.
   - All logs for a particular request have the ID of that request (no separate package required).
   - Each request has a `request_started` event logging the URL, HTTP method and user agent.
   - Each request also has a `request_finished` event logging the URL, HTTP method and response status code.
   - There is no need to `getChild` logger as each log entry records the logger name, file name, func name and even line where it was called.
   - Logs with only a text message become JSON like `{"event": "<message>"}`.
   - Other loggers (like those for Uvicorn and Elasticsearch) are not affected and continue to be unstructured.

## Checklist


      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
